docs/usage/containerd-registry-configuration.md

```diff
@@ -0,0 +1,105 @@
+
+##### Hosts Directory Pattern
+
+The hosts directory pattern is the new and recommended pattern for configuring registries. It is available starting `containerd@v1.5.0`. See the [upstream documentation](https://github.com/containerd/containerd/blob/main/docs/hosts.md).
+The above example in the hosts directory pattern looks as follows.
+The `/etc/containerd/config.toml` file has the following section:
+```toml
+version = 2
+
+[plugins."io.containerd.grpc.v1.cri".registry]
+   config_path = "/etc/containerd/certs.d"
+```
+
+The following hosts directory structure has to be created:
+```
+$ tree /etc/containerd/certs.d
+/etc/containerd/certs.d
+└── docker.io
+    └── hosts.toml
+```
+
+Finally, for the `docker.io` upstream registry, we configure a `hosts.toml` file as follows:
+```toml
+server = "https://registry-1.docker.io"
+
+[host."http://public-mirror.example.com"]
+  capabilities = ["pull", "resolve"]
+```
+
+> Note: The hosts directory pattern is available in `containerd` 1.5+.
+
+### Configuring `containerd` Registries for a Shoot
+
+> Note: The below-described functionality is provided by the `ContainerdRegistryHostsDir` feature gate in gardenlet.
+
+Gardener supports configuring `containerd` registries on a Shoot using the new [hosts directory pattern](https://github.com/containerd/containerd/blob/main/docs/hosts.md). For each Shoot Node, Gardener creates the `/etc/containerd/certs.d` directory and adds the following section to the containerd's `/etc/containerd/config.toml` file:
+```toml
+[plugins."io.containerd.grpc.v1.cri".registry] # gardener-managed
+   config_path = "/etc/containerd/certs.d"
+```
+
+This allows Shoot owners to use the [hosts directory pattern](https://github.com/containerd/containerd/blob/main/docs/hosts.md) to configure registries for containerd. To do this, the Shoot owners need to create a directory under `/etc/containerd/certs.d` that is named with the upstream registry host name. In the newly created directory, a `hosts.toml` file needs to be created. For more details, see the [hosts directory pattern section](#hosts-directory-pattern) and the [upstream documentation](https://github.com/containerd/containerd/blob/main/docs/hosts.md).
+
+### The registry-cache Extension
+
+[Configuring `containerd` registries for a Shoot](#configuring-containerd-registries-for-a-shoot) is not the recommended approach for configuring a pull through cache for a Shoot. There is a Gardener-native extension named [registry-cache](https://github.com/gardener/gardener-extension-registry-cache) that manages a pull through cache for a Shoot using the upstream [distribution/distribution](https://github.com/distribution/distribution) project.
```

Comment:
In the light of L76, it probably is for now? 😄

Reply:
You mean that the docs state that registry-cache is recommended approach for configuring a pull through cache but then the docs state that registry-cache is not recommended because it is under development yet.
I wanted to share that the pull through cache will be covered by the registry-cache but on the other side we don't have yet a production ready release.
If you want I can adapt such as:
```suggestion
[Configuring `containerd` registries for a Shoot](#configuring-containerd-registries-for-a-shoot) won't be the recommended approach for configuring a pull through cache for a Shoot in near future. There is a Gardener-native extension named [registry-cache](https://github.com/gardener/gardener-extension-registry-cache) that manages a pull through cache for a Shoot using the upstream [distribution/distribution](https://github.com/distribution/distribution) project.
```

Reply:
Yes, this sounds more consistent at least :)